snapshotter/internal/http_address_resolver.go

```diff
@@ -132,17 +108,27 @@ func queryAddress(writ http.ResponseWriter, req *http.Request) {
 	}
 
 	namespace := keys[0]
-	networkAddress, ok := store[namespace]
-	if !ok {
-		http.Error(writ, "Socket path not found", http.StatusNotFound)
+
+	fcClient, err := client.New(containerdSockPath + ".ttrpc")
+	if err != nil {
+		logger.WithError(err).Error("could not create firecracker client")
+		http.Error(writ, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+	defer fcClient.Close()
+
+	vmInfo, err := fcClient.GetVMInfo(context.Background(), &proto.GetVMInfoRequest{VMID: namespace})
```

Comment:
Go HTTP request objects have their own context associated, which defaults to `context.Background()`

https://pkg.go.dev/net/http#Request.Context

should we use that instead?

Reply:
Yeah thanks for the catch. I'll add that.